src/cmd/manage_nsfs.js

```diff
@@ -568,13 +568,31 @@ async function manage_account_operations(action, data, config_root, config_root_
         await delete_account(data, accounts_path, access_keys_path, config_root_backend);
     } else if (action === ACTIONS.LIST) {
         let accounts = await list_config_files(accounts_path, show_secrets);
+        accounts = filter_account_results(accounts, argv)
         if (!data.wide) accounts = accounts.map(item => ({ name: item.name }));
         write_stdout_response(ManageCLIResponse.AccountList, accounts);
     } else {
         throw_cli_error(ManageCLIError.InvalidAction);
     }
 }
 
+/**
+ * filter_account_results will filter the results based on supported given flags
+ * @param {any[]} accounts
+ * @param {{}} argv
+ */
+function filter_account_results(accounts, argv) {
+    //supported filters for list
+    const filters = _.pick(argv, ['uid', 'gid']); //if we add filters we should remove this comment and the comment 4 lines below (else)
+    for (const [key, val] of Object.entries(filters)) {
+        if (key === 'uid' || key === 'gid') {
+            accounts = accounts.filter(item => item.nsfs_account_config[key] === val);
```

Comment:
I would run on the accounts array once and check all the filters at the same time instead of running the loop filters.length times.